### PR TITLE
chore: Update dev deps

### PR DESCRIPTION
**Pull request by robcresswell**
Most of the dev deps here are quite outdated, so this patch batch updates them and fixes the minor linting issues

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -31,21 +31,21 @@
...
-    "ts-jest": "^26",
-    "ts-node": "^8",
-    "typescript": "~3.8.3"
+    "eslint": "^7.29.0",
+    "eslint-config-prettier": "^8.3.0",
+    "jest": "^27.0.5",
+    "prettier": "^2.3.1",
+    "ts-jest": "^27.0.3",
+    "ts-node": "^10.0.0",
+    "typescript": "~4.3.4"
```
Comment (darscan) [inline package.json:48]: Hmm.. I thought we were trapped on 3.x due to.. something weird that I can't remember

Comment (darscan) [reply package.json:48]: Also, is it wise to upgrade `typescript` without upgrading `tslib`?

Comment (robcresswell) [reply package.json:48]: I did consider this, but I'm not sure they're inherently coupled? I'm not changing the compile target

Comment (darscan) [reply package.json:48]: > something weird that I can't remember

But maybe not an issue for dep-graph. I think it's something to do with imports/exports and mocking

**Review by darscan — APPROVED** [review]

**Pull request merged**